drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,591 @@
+		SYS_LOG_DBG("failed reading chip id");
+		return -EIO;
+	}
+
+	if (!((chip_id == APDS9960_ID_1) || (chip_id == APDS9960_ID_2))) {
+		SYS_LOG_DBG("invalid chip id 0x%x", chip_id);
+		return -EIO;
+	}
+
+	if (!apds9960_setmode(dev, APDS9960_ALL, APDS9960_MODE_OFF)) {
+		SYS_LOG_DBG("ENABLE registered is not cleared");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ATIME_REG, APDS9960_DEFAULT_ATIME) < 0) {
+		SYS_LOG_DBG("Default integration time not set for ADC");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_WTIME_REG, APDS9960_DEFAULT_WTIME) < 0) {
+		SYS_LOG_DBG("Default wait time not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PPULSE_REG, APDS9960_DEFAULT_PROX_PPULSE) < 0) {
+		SYS_LOG_DBG("Default proximity ppulse not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_POFFSET_UR_REG, APDS9960_DEFAULT_POFFSET_UR) < 0) {
+		SYS_LOG_DBG("Default poffset UR not set ");
+		return -EIO;
+	}
+
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_POFFSET_DL_REG, APDS9960_DEFAULT_POFFSET_DL) < 0) {
+		SYS_LOG_DBG("Default poffset DL not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG1_REG, APDS9960_DEFAULT_CONFIG1) < 0) {
+		SYS_LOG_DBG("Default config1 not set ");
+		return -EIO;
+	}
+
+	if (!apds9960_setleddrive(dev, APDS9960_DEFAULT_LDRIVE)) {
+		SYS_LOG_DBG("LEDdrive not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setproximitygain(dev, APDS9960_DEFAULT_PGAIN)) {
+		SYS_LOG_DBG("proximity gain not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setambientlightgain(dev, APDS9960_DEFAULT_AGAIN)) {
+		SYS_LOG_DBG("ambient light gain not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setproxintlowthresh(dev, APDS9960_DEFAULT_PILT)) {
+		SYS_LOG_DBG("prox low threshold not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setproxinthighthresh(dev, APDS9960_DEFAULT_PIHT)) {
+		SYS_LOG_DBG("prox high threshold not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setlightintlowthresh(dev, APDS9960_DEFAULT_AILT)) {
+		SYS_LOG_DBG("light low threshold not set");
+		return -EIO;
+	}
+
+	if (!apds9960_setlightinthighthresh(dev, APDS9960_DEFAULT_AIHT)) {
+		SYS_LOG_DBG("light high threshold not set");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PERS_REG, APDS9960_DEFAULT_PERS) < 0) {
+		SYS_LOG_DBG("ALS interrupt persistence not set ");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG2_REG, APDS9960_DEFAULT_CONFIG2) < 0) {
+		SYS_LOG_DBG("clear diode saturation interrupt is not enabled");
+		return -EIO;
+	}
+
+	if ((apds9960_proxy_setup(dev, gain)) < 0) {
+		SYS_LOG_DBG("Failed to setup proxymity functionality");
+		return -EIO;
+	}
+
+	if ((apds9960_ambient_setup(dev, gain)) < 0) {
```

Comment:
apds_ambient_setup returns an errno hence this code should simply propagate the existing errno rather than replace it, I suspect the same comment applies to many of the preceding function calls but I've not checked them individually.